### PR TITLE
feat: allow tables to be resolved from a function

### PR DESCRIPTION
**Pull request by freshollie**
This PR adds support for #15. It uses a different approach to that specified in the issue, but achieves the same result. Please take a look @asarazan.

### CODE REVIEW

**Comment by asarazan** [comment]
This looks great! Side-note: I'm surprised TS doesn't have a built-in type for `T | Promise<T>`, and that there isn't a better solution for `isPromise` but yolo it's javascript :-D

Thanks for jumping on this so quickly!

File: .jest/configs/tables-function-async/jest-dynalite-config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const realTimeout = setTimeout;
```
Comment (corollari) [inline .jest/configs/tables-function-async/jest-dynalite-config.js:1]: Why rename an STL function?

Comment (freshollie) [reply .jest/configs/tables-function-async/jest-dynalite-config.js:1]: The test suite enables fake timers, so when jest-dynalite goes to execute the tables function the timeout has been mocked. The reference is just to allow the "sleep" function to work after the timers have been faked.

Comment (corollari) [reply .jest/configs/tables-function-async/jest-dynalite-config.js:1]: I see

**Pull request merged**